[New KB Article] Why aren't Supply Chain findings showing? (#1092)
returntocorp/semgrep-docs: docs/kb/semgrep-supply-chain/why-no-findings.md

---
description: Troubleshoot why findings for Semgrep Supply Chain are not showing.
tags:
  - Semgrep Supply Chain
---

> # Why aren't Supply Chain findings showing?

Reviewer: I assume (based on the content) that this referred to both reachable and unreachable findings - is this correct? And if so, do we want to mention it explicitly? Also, I assume this is applicable for CLI and SCP interfaces? Maybe we we want to mention that also.

Author: It's only applicable for SCP, so maybe I should specify. The reason I didn't is that non-SCP use of SSC is currently not considered a happy path, but I don't think it hurts to be clear.

This is all findings, yep. I could also change the title to say "any", maybe?

"Why aren't any Supply Chain findings showing in Semgrep Cloud Platform?" would be a possible revision.

Author: oh, wait a sec, I remember why I didn't specify. the reason is that a bunch of these possibilities apply to both - only the branch thing is specific to SCP.

Reviewer: "Why aren't any Supply Chain findings showing in Semgrep Cloud Platform" sounds good!

Author: Ooh okay. So it's probably worth either adding that as a section, or being super clear upfront. I'd actually incline to the former, I'll go back and add that now.

## Ensure compatibility

First, verify that your repository meets the basic requirements for Semgrep Supply Chain:

### Are the language and ecosystem supported?

Check the [Supported Languages table](/docs/supported-languages/#general-availability) for support for the project's language and ecosystem, as well as any ecosystem-specific requirements.

###  Is the supported lockfile present in an appropriate location with the expected name?

> Semgrep Supply Chain searches the parent directories of any files for a language and ecosystem for the nearest relevant lockfile. This typically means that monorepos are recognized by the presence of the lockfiles in subdirectories.

Author: Yeah, it's kind of a weird sentence. Do you have rephrasing suggestions? I also could just leave out "and ecosystem", which might be enough.

Do we really have to use "monorepositories"? I have never heard anyone utter that word. I'd probably switch it to "monolithic repositories" if I had to go formal, but `monorepo` is pretty much standard usage.

> Semgrep Supply Chain searches the parent directories of any files for a language and ecosystem for the nearest relevant lockfile. This typically means that monorepos are recognized by the presence of the lockfiles in subdirectories.

Author: revised by removing "and ecosystem" and changing to "monolithic repositories (monorepos)".

Only the lockfile names indicated in the [Supported Languages table](/docs/supported-languages/#general-availability) are recognized by Semgrep Supply Chain.

### Does the lockfile contain dependencies with pinned versions?

> If your dependency file is a [manifest file](/docs/semgrep-supply-chain/glossary/#manifest-file) and contains unpinned versions, Semgrep Supply Chain does not report vulnerabilities for the dependencies that are not pinned. An unpinned dependency may already be installed at a safe version for a particular [Advisory](https://semgrep.dev/docs/semgrep-supply-chain/glossary/#advisory).

Reviewer: ```suggestion
[introductory sentence about pinned versions] If your dependency file is a [manifest file](/docs/semgrep-supply-chain/glossary/#manifest-file) and contains unpinned versions, Semgrep Supply Chain does not report vulnerabilities for the dependencies that are not pinned. An unpinned dependency may already be installed at a safe version for a particular [Advisory](https://semgrep.dev/docs/semgrep-supply-chain/glossary/#advisory).
```

Maybe explain what pinned /unpinned versions are?

Author: Sure. I'll steal from our glossary :)

  - Check other organizations you belong to in Semgrep Cloud Platform to see if the results appear there.
  - If you are running `semgrep ci` locally, use `semgrep logout` and `semgrep login`, and ensure you log in to the desired Semgrep Cloud Platform organization.

### If the scan was a diff-aware (PR/MR) scan, was the lockfile modified? 

Semgrep Supply Chain only runs in [diff-aware scans](/docs/semgrep-ci/running-semgrep-ci-with-semgrep-cloud-platform/#diff-aware-scanning) if the lockfile was modified in the PR/MR.

If code is modified, but the lockfile is not, Supply Chain does not analyze the changes. Any code changes that might impact [reachability](/docs/semgrep-supply-chain/glossary/#reachability) will be identified on the next full scan.

### Did Semgrep scan a ref other than your default branch, or does your default branch have a less common name?

Currently, Semgrep Supply Chain only displays findings on one of the following branches in Semgrep Cloud Platform:

* The repository's default branch, if that information is available. This information is typically available for CI scans performed through GitHub Actions.

> * One of a set of standard default branch names, including `develop` (or `development`), `main`, `master`, and `trunk`.

Reviewer: ```suggestion
* One of a set of standard default branch names, such as: 
    * `develop` (or `development`)
    *`main`
    * `master`
    `trunk`.
```

If a scan runs on a different branch, findings do not show in Semgrep Cloud Platform. This prevents vulnerability findings from persisting incorrectly after they have been resolved on the repository's primary branch.

## Additional references

If the project uses Java and Apache Maven with `pom.xml`, see [Setting up SSC scans for specific project management tools:
Apache Maven (Java)](/docs/semgrep-supply-chain/getting-started/#apache-maven-java).

## If you're still having trouble

> If you've addressed these issues but are still not seeing vulnerability findings, or if you need assistance setting up Semgrep Supply Chain for your projects, such as handling lockfile naming or addressing parsing issues, please [reach out for help](docs/support/).

Author: Honestly, I think it's fine here (it's not asking someone to please click something, they have to actually make an effort), but I don't think it's necessary, so I'm cool with removing it.

### Does the lockfile contain dependencies with exact versions?

If your dependency file is a [manifest file](/docs/semgrep-supply-chain/glossary/#manifest-file) and does not specify exact (pinned) versions for all dependencies, Semgrep Supply Chain does not report vulnerabilities for the dependencies that are not pinned. An unpinned dependency may already be installed at a safe version for a particular [Advisory](https://semgrep.dev/docs/semgrep-supply-chain/glossary/#advisory).

Pinned dependencies can be analyzed even if the file contains other unpinned dependencies. Manifest files can also be helpful to determine whether a dependency is [transitive](/docs/semgrep-supply-chain/glossary/#transitive-or-indirect-dependency).

## Check scan status and result location

Next, ensure that the scan was successful and sent results to the expected location.

### Did the scan run successfully?

If the scan did not complete, or failed when trying to send results, the dependency information will not be available.

Review the logs from the scan and determine whether it was successful, or ran into an issue. Common issues include:

> * The lockfile could not be parsed successfully. The CI output should point to the line where the error occurred.

Author: That's a nice idea! Do you have a good example I can add? I haven't run into this so far.

In regards to the CTA, there is one at the bottom of the article, will that work? I could also put a link to move down to that section here, since if you get a parsing error the rest is probably moot.

> * The lockfile could not be parsed successfully. The CI output should point to the line where the error occurred.

Reviewer 2: <img width="882" alt="Screen Shot 2023-06-29 at 12 59 11 PM" src="https://github.com/returntocorp/semgrep-docs/assets/100232702/49bce721-3211-44a2-af12-8df382551411">

something like this maybe?

> * The lockfile could not be parsed successfully. The CI output should point to the line where the error occurred.

Reviewer 2: For CTA, I like the latter! Yeah, let's just get folks immediately to the CTA + reach out for help phase if they're truly stuck

> * The lockfile could not be parsed successfully. The CI output should point to the line where the error occurred.

Author: Thanks! I'll get those updates done now.

---
description: Troubleshoot why findings for Semgrep Supply Chain are not showing.
tags:
  - Semgrep Supply Chain
---
# Why aren't Supply Chain findings showing?

## Ensure compatibility

First, verify that your repository meets the basic requirements for Semgrep Supply Chain:

### Are the language and ecosystem supported?

> Check the [Supported Languages table](/docs/supported-languages/#general-availability) for support for the project's language and ecosystem, as well as any ecosystem-specific requirements.

Reviewer 3: ```suggestion
Check the [Supported Languages table](/docs/supported-languages/#general-availability) to verify the project's language and ecosystem are supported, as well as any ecosystem-specific requirements.
```

Author: How about "to verify support for" - keep it short but remove the weird phrasing (good catch, ty).

###  Is the supported lockfile present in an appropriate location with the expected name?

> Semgrep Supply Chain searches the parent directories of any files for a language for the nearest relevant lockfile. This typically means that monolithic repositories (monorepos) are recognized by the presence of the lockfiles in subdirectories.

Reviewer 3: I'm a little confused at the overall message here. Should we be mentioning the appropriate place for lockfiles to be placed?

Semgrep Supply Chain only recognizes the lockfile names indicated in the [Supported Languages table](/docs/supported-languages/#general-availability).

### Does the lockfile contain dependencies with exact versions?

> If your dependency file is a [manifest file](/docs/semgrep-supply-chain/glossary/#manifest-file) and does not specify exact (pinned) versions for all dependencies, Semgrep Supply Chain does not report vulnerabilities for the dependencies that are not pinned. An unpinned dependency may already be installed at a safe version for a particular [Advisory](https://semgrep.dev/docs/semgrep-supply-chain/glossary/#advisory).

Reviewer 3: For "An unpinned dependency may already be installed at a safe version for a particular Advisory", what do we mean by this? Do we mean that this is why we don't check for vulns for unpinned dependencies?

Author: yep! do you have a suggestion to make that more clear?

* The repository's default branch, if that information is available. This information is typically available for CI scans performed through GitHub Actions.
* One of a set of standard default branch names, such as:
  * `develop` (or `development`)
  * `main`
  * `master`
  * `trunk`

If a scan runs on a different branch, findings do not show in Semgrep Cloud Platform. This prevents vulnerability findings from persisting incorrectly after they have been resolved on the repository's primary branch.

### Are all findings in the scan unreachable vulnerabilities?

By default, Semgrep Cloud Platform shows only [reachable](/docs/semgrep-supply-chain/glossary/#reachability) vulnerabilities in the UI. 

To see all vulnerabilities, select all boxes under the "Exposure" filter.

![Exposure filter with all boxes selected](/img/kb/ssc-vuln-filter.png)

## Additional references

If the project uses Java and Apache Maven with `pom.xml`, see [Setting up SSC scans for specific project management tools:
Apache Maven (Java)](/docs/semgrep-supply-chain/getting-started/#apache-maven-java).

## If you're still having trouble

> If you've addressed these issues but are still not seeing vulnerability findings, or if you need assistance setting up Semgrep Supply Chain for your projects, such as handling lockfile naming or addressing parsing issues, [reach out for help](docs/support/).

Reviewer 3: ```suggestion
If you've addressed these issues but are still not seeing vulnerability findings, or if you need assistance setting up Semgrep Supply Chain for your projects, such as handling lockfile naming or addressing parsing issues, [reach out for help](/docs/support/).
```
